From 464758b2d3a1af8fad402acdbb5f3a157b4accab Mon Sep 17 00:00:00 2001
From: Liu Ying <Ying.liu@freescale.com>
Date: Tue, 10 Jul 2012 12:00:17 +0800
Subject: [PATCH 1730/2463] ENGR00216268 Ov5642 camera:Fix green line issue
 for 2 modes

This patch fix green line issue on captured frames for
720x480p@30 and 720x576p@30 modes by changing register
0x302c's bit[6:5] to 0'11 to enhance output driving
capability to 4x.

Signed-off-by: Liu Ying <Ying.Liu@freescale.com>
---
 drivers/media/video/mxc/capture/ov5642.c |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/drivers/media/video/mxc/capture/ov5642.c b/drivers/media/video/mxc/capture/ov5642.c
index 2488658..31dc279 100644
--- a/drivers/media/video/mxc/capture/ov5642.c
+++ b/drivers/media/video/mxc/capture/ov5642.c
@@ -1699,6 +1699,7 @@ static struct reg_value ov5642_setting_30fps_NTSC_720_480[] = {
 	{0x5013, 0x00, 0, 0}, {0x519e, 0x00, 0, 0}, {0x5086, 0x00, 0, 0},
 	{0x5087, 0x00, 0, 0}, {0x5088, 0x00, 0, 0}, {0x5089, 0x00, 0, 0},
 	{0x302b, 0x00, 0, 0}, {0x3621, 0x87, 0, 0}, {0x3a00, 0x78, 0, 0},
+	{0x302c, 0x60, 0x60, 0},
 };
 
 static struct reg_value ov5642_setting_30fps_PAL_720_576[] = {
@@ -1868,6 +1869,7 @@ static struct reg_value ov5642_setting_30fps_PAL_720_576[] = {
 	{0x5013, 0x00, 0, 0}, {0x519e, 0x00, 0, 0}, {0x5086, 0x00, 0, 0},
 	{0x5087, 0x00, 0, 0}, {0x5088, 0x00, 0, 0}, {0x5089, 0x00, 0, 0},
 	{0x302b, 0x00, 0, 0}, {0x3621, 0x87, 0, 0}, {0x3a00, 0x78, 0, 0},
+	{0x302c, 0x60, 0x60, 0},
 };
 
 static struct reg_value ov5642_setting_15fps_720P_1280_720[] = {
-- 
1.7.7.4

